### DISCUSSION TITLE
Add jwt auth to swagger

### DISCUSSION THREAD

**Discussion by Master-Y0da** [fastapi-users/fastapi-users#593] [Q&A] [answered]
When you set jwt authentication...swagger does'nt allow anymore to try your endpoints, since you should include the token in request...how can I add it ?

**Comment by frankie567** [comment, marked as answer]
Did you try to authenticate with the Authenticate button at the top? 

<img width="1565" alt="Capture d’écran 2021-04-08 à 17 50 01" src="https://user-images.githubusercontent.com/1144727/114057482-e33cec80-9892-11eb-9d0a-a80fc065cb38.png">

In the window that appear, just type the e-mail (username field) and the password of your user and Swagger should automatically put the token when you try requests.

**Reply by Master-Y0da** [reply]
@frankie567 Yeah I already try that...and throws this error [error cap](http://i.imgur.com/zgHAQOZ.png)

**Reply by frankie567** [reply]
Which version of `fastapi-users` are you using?

**Reply by Master-Y0da** [reply]
3.0.6

**Reply by frankie567** [reply]
Ok, I think you really should upgrade to the latest versions, there are lot of fixes and improvements that'll make your life easier.

Now, I would be curious to see the line instantiating your JWTAuthentication class. Especially check the `tokenUrl` argument, it should match your login route so that it works inside Swagger:

```py
jwt_authentication = JWTAuthentication(
    secret=SECRET, lifetime_seconds=3600, tokenUrl="/auth/jwt/login"
)
```

**Reply by Master-Y0da** [reply]
That was the problem!! ...I complete forgot to add tokenUrl...thank you man...you are great!!
